src/lib/convert.ts

```diff
@@ -1,30 +1,68 @@
-import Handlebars from 'handlebars';
+import Handlebars, { Exception } from 'handlebars';
+import { String } from 'shelljs';
 
 const convert = input => {
   const transpiler = new Transpiler(input);
   const output = transpiler.toString();
   return output;
 };
 
+class Contexts {
+  private _contexts: any[];
+
+  constructor() {
+    this._contexts = [];
+  }
+
+  push(s) {
+    this._contexts.push(s);
+  }
+
+  getScope(depth) {
+    return this._contexts[depth - 1];
+  }
+
+  getScopedVariable(depth, varName) {
+    const sc = this.getScope(depth);
+    if (typeof sc === 'undefined') {
+      return varName;
+    }
+    // if context has 2 variables as in {each foo |k, v|}
+    // we dont have to bind variables
+    // as user will use {{v.attribute}} etc.
+    if (sc.length === 2) return varName;
```

Comment:
yes, maybe, I thought about that too, but used this as a shortcut for now :) just to pass the tests

please bear with me for a lengthy comment.

what I am thinking is say you have a variable that is on the outer scope something like this
```
hbs = require('handlebars')
var source = `
{{#each posts as |post|}}
    <li>{{post.id}}</li>
    {{#each comments as |comment|}}
        <li>{{post.id}} - {{comment.title}}</li>
    {{/each}}
{{/each}}
`
var template = hbs.compile(source)
data = {
    posts: [{id: 1, comments: [{title: "foo"}]}]
}
console.log(template(data))
```

in this case when we see, `{{post.id}}` we should look into outer scope as well. so we have to iterate until the very first scope if we have a match. What do you think ?

Reply:
When dealing with the `|value key|` part, we don't even have to add the to the scope, since the program will use `value.foo` explicitly. We don't introduce new variables in the output template that we have to prepend, so in this case we can ignore those at all, right?

Reply:
I think, we need more test cases, for these. Probably you are right but I am not sure. Though we'll have to translate this
`{{#each comments as |comment|}}`
 to something like 
`{% for comments_i in post.comments %}`

so we'll have to look to the upper scope - or maybe not :) - but I am not sure without the test cases. 

but well thats why there is a shortcut `if (sc.length === 2)` for now, it will most likely fail on more complicated test cases and will need better logic. Which is hard to foresee.

Reply:
Nope,
`{{#each comments as |comment|}}`
will just be
`% for comment in post.comments %}`

Reply:
yes :) though I mean `comments` will need to be `post.comments`, and to find if its comments or post.comments we'll probably have to look to the upper scope.

Reply:
Ah yes, when you use `|value|` you can still use the normal scope as well... that's going to be a complex process of lookup and elimination.

* first, resolve the scope for the variable (using ../)
* then, check if the first part is present in any of the parent scopes
* if not, prepend it with the currently resolved scope (from step 1)